docs/nix-develop.md

```diff
@@ -0,0 +1,206 @@
+
+- located `nix/rust.nix`
+```nix
+    rustOverlaySet = mkOption {
+      type = types.listOf types.str;
+      default = [
+        "rustc"
+        "cargo"
+        "rustfmt-preview"
+        "<add other package which you can found in package-list>"
+      ];
+      description = "Which rust tools to pull from the rust overlay
+      https://github.com/oxalica/rust-overlay/blob/master/manifests/profiles.nix";
+    };
+```
+### Add rustPackages from nixpkgs channel
+
+- located `nix/rust.nix` and the default nix expression like this  `rustPackages.clippy`. In this cause, most of rust toolchain installed from `rustOverly`
+
+```nix
+    rustPackagesSet = mkOption {
+      type = types.listOf types.str;
+      default = [
+        "clippy"
+        "<add custom Rust toolchain package>"
+      ];
+      description = "Which rust tools to pull from the nixpkgs channel package set
+      check valid package from https://search.nixos.org/packages?channel=unstable";
+      ";
+    };
+```
+
+### Add nixpkgs package to current Environment
+
+- located `nix/rust.nix`
+
+first search any package in [search-package](https://search.nixos.org/packages?channel=unstable&) then add the name to 
+
+```nix
+     devshell.packages = map (tool: cfg.rustPackages.${tool}) cfg.rustPackagesSet
+      ++ map (tool: cfg.rustOverlay.${tool}) cfg.rustOverlaySet ++ (with pkgs;[
+      #custom nixpkgs packages
+      rustracer
+      <add package here>
+    ]);
+ ```
+
+## Reload all of envs when you changed something
+
+- normally, we can re-enter `nix-shell` to reload envrs. for convinet
+
+### Recommended way -> Direnv
+
+ Install direnv by Nix
+
+    nix-env -i direnv
+
+and hook direnv to your bash [direnv-hook](https://direnv.net/docs/hook.html)
+
+- using `direnv-allow`  or `direnv-deny` to enable or disable load `nix-shell` in current path
+
+- using `direnv-reload` to reload `nix-shell`
+
+- Also, you can put as follows to `~/.config/direnv/direnvrc` watching the envrs change that could be reload automatically.
+
+```conf
+use_flake() {
+  watch_file flake.nix
+  watch_file flake.lock
+  eval "$(nix print-dev-env)"
+}
+```
+
+### Add direnv support with Emacs
+
+- <https://github.com/purcell/envrc>
+
+Install direnv to Doom emacs, Exmaple
+
+```elisp
+(package! envrc :recipe (:host github :repo "purcell/envrc"))
+(use-package! envrc
+:hook (after-init . envrc-global-mode)
+)
+```
+ then `M-x` typing `envrv-<**>` check related commands similar to native direnv-commands
+
+### write custom commands or environment variables
+
+- locaed `nix/commands`
+
+- Add command setting as following: 
+ Example
+
+```toml
+[[commands]]
+name = ""
+command = "cargo build --manifest-path=./rust_src/remacs-bindings/Cargo.toml"
+help = "cargo build remacs-bindings"
+category = "rust-build"
+```
+
+- Add env setting as following: 
+
+```toml
+[[env]]
+name = "TEST"
+vale = "/bin/test"
+#prefix = "$( cd "$(dirname "$\{\BASH_SOURCE [ 0 ]}")"; pwd )" can be prefix 
+```
+
+
+
```

Comment:
```suggestion
```

Reply:
Sorry, wait a moment. Let me rewrite the doc.